// toastfly/bundles/toastfly.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('toastfly', ['exports', '@angular/core'], factory) :
    (global = global || self, factory(global.toastfly = {}, global.ng.core));
}(this, (function (exports, i0) { 'use strict';

    var ToastflyService = /** @class */ (function () {
        function ToastflyService() {
        }
        ToastflyService.prototype.positive = function (msg) {
            var toast = document.getElementById('toast');
            toast.style.backgroundColor = 'green';
            toast.innerHTML = msg;
            toast.className = "show";
            setTimeout(function () {
                toast.className = toast.className.replace("show", "");
            }, 3000);
        };
        ToastflyService.prototype.negative = function (msg) {
            var toast = document.getElementById('toast');
            toast.style.backgroundColor = 'red';
            toast.innerHTML = msg;
            toast.className = "show";
            setTimeout(function () {
                toast.className = toast.className.replace("show", "");
            }, 3000);
        };
        return ToastflyService;
    }());
    ToastflyService.ɵprov = i0.ɵɵdefineInjectable({ factory: function ToastflyService_Factory() { return new ToastflyService(); }, token: ToastflyService, providedIn: "root" });
    ToastflyService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    ToastflyService.ctorParameters = function () { return []; };

    var ToastflyComponent = /** @class */ (function () {
        function ToastflyComponent() {
        }
        ToastflyComponent.prototype.ngOnInit = function () {
        };
        return ToastflyComponent;
    }());
    ToastflyComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'lib-toastfly',
                    template: "\n\n  <div id=\"toast\"></div>\n  ",
                    styles: ["#toast{border-radius:2px;bottom:30px;color:#fff;left:50%;margin-left:-125px;min-width:250px;padding:16px;position:fixed;text-align:center;visibility:hidden;z-index:1}#toast.show{-webkit-animation:fadein .5s,fadeout .5s 2.5s;animation:fadein .5s,fadeout .5s 2.5s;visibility:visible}@-webkit-keyframes fadein{0%{bottom:0;opacity:0}to{bottom:30px;opacity:1}}@keyframes fadein{0%{bottom:0;opacity:0}to{bottom:30px;opacity:1}}@-webkit-keyframes fadeout{0%{bottom:30px;opacity:1}to{bottom:0;opacity:0}}@keyframes fadeout{0%{bottom:30px;opacity:1}to{bottom:0;opacity:0}}"]
                },] }
    ];

    var ToastflyModule = /** @class */ (function () {
        function ToastflyModule() {
        }
        return ToastflyModule;
    }());
    ToastflyModule.decorators = [
        { type: i0.NgModule, args: [{
                    declarations: [ToastflyComponent],
                    imports: [],
                    exports: [ToastflyComponent]
                },] }
    ];

    /*
     * Public API Surface of toastfly
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ToastflyComponent = ToastflyComponent;
    exports.ToastflyModule = ToastflyModule;
    exports.ToastflyService = ToastflyService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
