// toastfly/fesm2015/toastfly.js
import { ɵɵdefineInjectable, Injectable, Component, NgModule } from '@angular/core';

class ToastflyService {
    constructor() { }
    positive(msg) {
        var toast = document.getElementById('toast');
        toast.style.backgroundColor = 'green';
        toast.innerHTML = msg;
        toast.className = "show";
        setTimeout(() => {
            toast.className = toast.className.replace("show", "");
        }, 3000);
    }
    negative(msg) {
        var toast = document.getElementById('toast');
        toast.style.backgroundColor = 'red';
        toast.innerHTML = msg;
        toast.className = "show";
        setTimeout(() => {
            toast.className = toast.className.replace("show", "");
        }, 3000);
    }
}
ToastflyService.ɵprov = ɵɵdefineInjectable({ factory: function ToastflyService_Factory() { return new ToastflyService(); }, token: ToastflyService, providedIn: "root" });
ToastflyService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
ToastflyService.ctorParameters = () => [];

class ToastflyComponent {
    ngOnInit() {
    }
}
ToastflyComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-toastfly',
                template: `

  <div id="toast"></div>
  `,
                styles: ["#toast{border-radius:2px;bottom:30px;color:#fff;left:50%;margin-left:-125px;min-width:250px;padding:16px;position:fixed;text-align:center;visibility:hidden;z-index:1}#toast.show{-webkit-animation:fadein .5s,fadeout .5s 2.5s;animation:fadein .5s,fadeout .5s 2.5s;visibility:visible}@-webkit-keyframes fadein{0%{bottom:0;opacity:0}to{bottom:30px;opacity:1}}@keyframes fadein{0%{bottom:0;opacity:0}to{bottom:30px;opacity:1}}@-webkit-keyframes fadeout{0%{bottom:30px;opacity:1}to{bottom:0;opacity:0}}@keyframes fadeout{0%{bottom:30px;opacity:1}to{bottom:0;opacity:0}}"]
            },] }
];

class ToastflyModule {
}
ToastflyModule.decorators = [
    { type: NgModule, args: [{
                declarations: [ToastflyComponent],
                imports: [],
                exports: [ToastflyComponent]
            },] }
];

/*
 * Public API Surface of toastfly
 */

/**
 * Generated bundle index. Do not edit.
 */

export { ToastflyComponent, ToastflyModule, ToastflyService };
